src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -135,24 +135,23 @@ public void startTask(String tableName, String topic, int partition) {
     // the table should be present before opening a channel so lets do a table existence check here
     createTableIfNotExists(tableName);
 
-    createOrUpdatePartitionsToChannelCache(partitionChannelKey, tableName);
+    createStreamingChannelForTopicPartition(partitionChannelKey, tableName);
   }
 
-  /* We will always open a new channel. This will either update the TopicPartitionChannel's streamingChannel or will create a new instance of TopicPartitionChannel. */
-  private void createOrUpdatePartitionsToChannelCache(
+  /**
+   * Always opens a new channel and creates a new instance of TopicPartitionChannel.
+   *
+   * <p>This is essentially a blind write to partitionsToChannel. i.e we dont check if it is present
+   * or not.
+   */
+  private void createStreamingChannelForTopicPartition(
       final String partitionChannelKey, final String tableName) {
     // Always open a new channel.
     SnowflakeStreamingIngestChannel partitionChannel =
         openChannelForTable(partitionChannelKey, tableName);
-    if (partitionsToChannel.containsKey(partitionChannelKey)) {
-      // we just update the channel instance for this partition
-      partitionsToChannel.get(partitionChannelKey).updateStreamingIngestChannel(partitionChannel);
-    } else {
-      // Add channel and related metadata to partitionsToChannel map
-      TopicPartitionChannel topicPartitionChannel =
-          new TopicPartitionChannel(partitionChannel, this.kafkaRecordErrorReporter);
-      partitionsToChannel.put(partitionChannelKey, topicPartitionChannel);
-    }
+    TopicPartitionChannel topicPartitionChannel =
+        new TopicPartitionChannel(partitionChannel, this.kafkaRecordErrorReporter);
+    partitionsToChannel.put(partitionChannelKey, topicPartitionChannel);
```

Comment:
Nit, just inline  the above `TopicPartitionChannel` creation here.

Reply:
Taken care in next PR.